images/win/scripts/SoftwareReport/SoftwareReport.Tools.psm1

```diff
@@ -0,0 +1,233 @@
+    $zstdVersion = $Matches.Version
+    return "zstd $zstdVersion"
+}
+
+function Get-AzureCLIVersion {
+    $azureCLIVersion = $(az version) | ConvertFrom-Json | Foreach{ $_."azure-cli" }
+    return "Azure CLI $azureCLIVersion"
+}
+
+function Get-AzureDevopsExtVersion {
+    $azureDevExtVersion = (az version | ConvertFrom-Json | Foreach{ $_."extensions" })."azure-devops"
+    return "Azure DevOps CLI extension $azureDevExtVersion"
+}
+
+function Get-AWSCLIVersion {
+    $(aws --version) -match "aws-cli\/(?<version>\d+\.\d+\.\d+)" | Out-Null
+    $awscliVersion = $Matches.Version
+    return "AWS CLI $awscliVersion"
+}
+
+function Get-AWSSAMVersion {
+    $(sam --version) -match "version (?<version>\d+\.\d+\.\d+)" | Out-Null
+    $awssamVersion = $Matches.Version
+    return "AWS SAM CLI $awssamVersion"
+}
+
+function Get-AlibabaCLIVersion {
+    $(aliyun --version | Select-String "Alibaba Cloud Command Line Interface") -match "(?<version>\d+\.\d+\.\d+)" | Out-Null
+    $alicliVersion = $Matches.Version
+    return "Alibaba CLI $alicliVersion"
+}
+
+function Get-CloudFoundryVersion {
+    $(cf version) -match  "(?<version>\d+\.\d+\.\d+)" | Out-Null
+    $cfVersion = $Matches.Version
+    return "Cloud Foundry CLI $cfVersion"
+}
+
+function Get-HubVersion {
+    ($(hub version) | Select-String -Pattern "hub version") -match "hub version (?<version>\d+\.\d+\.\d+)" | Out-Null
+    $hubVersion = $Matches.Version
+    return "Hub CLI $hubVersion"
+}
+
+$browsers = @{
+    chrome = @{
+        Name="Google Chrome";
+        File="chrome.exe"
+    };
+    edge = @{
+        Name="Microsoft Edge";
+        File="msedge.exe"
+    };    
+    firefox = @{
+        Name="Mozilla Firefox";
+        File="firefox.exe"
+    }
+}
+
+$webDrivers = @{
+    chrome = @{
+        Name="Chrome Driver";
+        Path="C:\SeleniumWebDrivers\ChromeDriver"
+    };
+    edge = @{
+        Name="Microsoft Edge Driver";
+        Path="C:\SeleniumWebDrivers\EdgeDriver"
+    };
+    firefox = @{
+        Name="Gecko Driver";
+        Path="C:\SeleniumWebDrivers\GeckoDriver"
+    };
+    iexplorer = @{
+        Name="IE Driver";
+        Path="C:\SeleniumWebDrivers\IEDriver"
+    }
+}
+
+function Get-BrowserVersion {
+    param(
+        [string] $Browser
+    )
+    $browserName = $browsers.$Browser.Name
+    $browserFile = $browsers.$Browser.File
+    $registryKey = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\$browserFile"
+    $browserVersion = (Get-Item (Get-ItemProperty $registryKey)."(Default)").VersionInfo.FileVersion
+    return "$browserName $browserVersion"
+}
+
+function Get-SeleniumWebDriverVersion {
+    param(
+        [string] $Driver
+    )
+    $driverName = $webDrivers.$Driver.Name
+    $driverPath = $webDrivers.$Driver.Path
+    $versionFileName = "versioninfo.txt";
+	$webDriverVersion = Get-Content -Path "$driverPath\$versionFileName"
+    return "$driverName $webDriverVersion"
+}
```

Comment:
Let's move these functions to the separate helper `SoftwareReport.Browsers.psm1`

Reply:
Done